src/api/mocks/src/config.ts

```diff
@@ -0,0 +1,258 @@
+import type { EndpointDependencies } from '@/api/mocks/index'
+export default <T extends EndpointDependencies>(_deps: T) => () => {
+  return {
+    headers: {},
+    body: {
```

Comment:
**Suggestion**: Can we please hook up the body with the appropriate type (i.e. `ClientConfigInterface`) so we get type checking in mock data generation? That would be a significant advantage over the current JSON file format.

Reply:
HTTP responses aren't typed. In my mind any typing should happen in the application layer, not in the HTTP mocks.

Reply:
I really disagree with this assessment. This is not some closed-off box where we can’t have types (as is the case for actual HTTP responses being surfaced through fetch, etc.). We can easily have types because we already have the definitions and because we write the JavaScript for the mock definitions. We would have a much easier time to spot mistakes in the mock data with this. And there isn’t an better place for this either: once this mock data is served, we’re looking at a running application and are well past a stage where the TypeScript compiler can tell us anything.

Having types here is strictly better than not having them on account of "HTTP responses aren't typed".

Reply:
It’s a suggestion and I’m happy to propose the change in a follow-up PR.

That being said, this can be easily accomplished by adding `: { headers: {}, body: ClientConfigInterface }` as the return type for the inner function and importing the appropriate type like so:

```ts
import type { ClientConfigInterface } from '@/store/modules/config/config.types'
```

Reply:
FYI I tried this out:

> That being said, this can be easily accomplished by adding : { headers: {}, body: ClientConfigInterface } as the return type for the inner function and importing the appropriate type like so:

```javascript
import type { ClientConfigInterface } from '@/store/modules/config/config.types'
```

And all the tests/linting failed.

I can fix that up by adding `any` to `MockResponder`, but I'm trying to avoid using `any` if I can help it, so I'm getting into rabbit hole territory I feel.

On a separate point and already mentioned point, I'm still unsure whether I want the functions that I would like to return bare untyped JSON (i.e. `Record<string, unknown>`) return an application type instead. I hope to know a little more once I work on the testing side of this.

Right now, I'm 'pretty sure' I want them to return bare untyped JSON. I may change my mind on this later, but even then I don't think I would type them with our frontend application types. So of right now this is where I am.

All in all, even if we were ok with `any`s, I'd very much rather leave them untyped, at least for the moment.

Reply:
> And all the tests/linting failed.

Exactly _one_ TypeScript error arises which propagates into the tests failing because virtually all of them import this file.

Avoiding `any` for a piece of data that we won’t read from in application logic in trade for having typed mock definitions is not worth it, but anyway.

Reply:
Appreciate your take on this, and I'm super happy to revisit at a later date. I want to try some more things out before I decide exactly what I would like and we can discuss further. Just so I understand correctly - is this convo resolved in the meantime?